src/net/dns.rs

```diff
@@ -42,33 +82,105 @@ pub(crate) async fn prune_dns_cache(context: &Context) -> Result<()> {
     Ok(())
 }
 
-/// Looks up the hostname and updates DNS cache
-/// on success.
+/// Map from hostname to IP addresses.
+static LOOKUP_HOST_CACHE: Lazy<std::sync::Mutex<HashMap<String, Vec<IpAddr>>>> =
```

Comment:
This could get a comment that we're using a non-async Mutex, which means that it must not be held across await points.